Source/JavaScriptCore/inspector/agents/InspectorDebuggerAgent.cpp

```diff
@@ -1961,6 +1959,19 @@ void InspectorDebuggerAgent::clearAsyncStackTraceData()
     didClearAsyncStackTraceData();
 }
 
+bool InspectorDebuggerAgent::BlackboxedScript::matches(const String& url)
+{
+    if (url.isEmpty())
+        return false;
+
+    if (!m_urlSearcher) {
+        auto searchType = isRegex ? ContentSearchUtilities::SearchType::Regex : ContentSearchUtilities::SearchType::ExactString;
+        auto searchCaseSensitive = caseSensitive ? ContentSearchUtilities::SearchCaseSensitive::Yes : ContentSearchUtilities::SearchCaseSensitive::No;
+        m_urlSearcher = ContentSearchUtilities::createSearcherForString(this->url, searchType, searchCaseSensitive);
+    }
+    return ContentSearchUtilities::searcherMatchesText(*m_urlSearcher, url);
```

Comment:
we could have a `HashSet` of known matching URLs, but i dont think it's that common for multiple `.js` to have the same URL in a single page session